CMakeLists.txt

```diff
@@ -735,3 +736,21 @@ if (ENABLE_FUZZING)
 
   add_subdirectory(fuzz)
 endif ()
+
+if (ENABLE_HEADERS_CHECK)
+  ## add headers sanity check project that includes all headers independently
+  set(check_headers_dir "${PROJECT_BINARY_DIR}/check-headers")
+  file(GLOB_RECURSE headers_to_check
+    ${PROJECT_BINARY_DIR}/*.hpp
+    ${PROJECT_SOURCE_DIR}/include/*.hpp)
+  foreach(header ${headers_to_check})
+    get_filename_component(filename ${header} NAME_WE)
+    set(filename "${check_headers_dir}/${filename}.cpp")
+    if (NOT EXISTS ${filename})
+      file(WRITE ${filename} "#include \"${header}\"\n")
+    endif()
+    list(APPEND sources ${filename})
+  endforeach()
+  add_library(check-headers STATIC EXCLUDE_FROM_ALL ${sources})
```

Comment:
removed, makes no sense. Just wanted to double check that it will not be included in `all`